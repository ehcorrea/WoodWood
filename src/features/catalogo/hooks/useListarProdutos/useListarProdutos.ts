import { useQuery } from '@tanstack/react-query';
import { useEffect } from 'react';
import { Alert } from 'react-native';

import { listarProdutos } from '@/services/api';

export function useListarProdutos() {
  const { data, isLoading, refetch, isError } = useQuery({
    queryKey: ['listarProdutos'],
    queryFn: listarProdutos,
  });

  useEffect(() => {
    if (isError) {
      Alert.alert('Error', 'Tentar Novamente?', [
        { text: 'Sim', onPress: () => refetch() },
        { text: 'NÃ£o' },
      ]);
    }
  }, [isError]);

  return { produtos: data, isLoading };
}
