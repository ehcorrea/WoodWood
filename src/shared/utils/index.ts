export * from './carrinho/carrinho';
export * from './responsive/responsive';
