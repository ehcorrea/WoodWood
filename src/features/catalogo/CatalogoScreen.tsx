import { ActivityIndicator } from 'react-native';

import { Spacing } from '@/shared/components';

import { useListarProdutos } from './hooks';
import { Categorias, ListaDeProdutos } from './components';

import * as S from './CatalogoScreen.styles';

export default function CatalogoScreen() {
  const { isLoading } = useListarProdutos();

  if (isLoading) {
    return <ActivityIndicator size="large" testID="loading" />;
  }

  return (
    <S.Wrapper>
      <Categorias />
      <Spacing y={10} />
      <ListaDeProdutos />
    </S.Wrapper>
  );
}
