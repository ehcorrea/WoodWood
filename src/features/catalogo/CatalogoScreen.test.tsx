import { View as MockedView } from 'react-native';

import { render } from '@/test/utils';

import * as useListarProdutos from './hooks/useListarProdutos/useListarProdutos';

import CatalogoScreen from './CatalogoScreen';

jest.mock('./components', () => ({
  Categorias: () => <MockedView testID="categorias" />,
  ListaDeProdutos: () => <MockedView testID="listaDeProdutos" />,
}));

const mockUseListarProdutos = jest.fn();

const setup = (isLoading = false) => {
  jest.spyOn(useListarProdutos, 'useListarProdutos').mockImplementation(() => {
    mockUseListarProdutos();
    return { isLoading };
  });
  return render(<CatalogoScreen />);
};

describe('<CatalogoScreen/>', () => {
  describe('quando renderizado', () => {
    test('e em loading', () => {
      const container = setup(true);
      expect(mockUseListarProdutos).toHaveBeenCalled();
      expect(container.getByTestId('loading')).toBeTruthy();
      expect(container.queryByTestId('categorias')).toBeFalsy();
      expect(container.queryByTestId('listaDeProdutos')).toBeFalsy();
      expect(container).toMatchSnapshot();
    });
  });
  test('quando renderizado', () => {
    const container = setup();
    expect(container.queryByTestId('loading')).toBeFalsy();
    expect(container.getByTestId('categorias')).toBeTruthy();
    expect(container.getByTestId('listaDeProdutos')).toBeTruthy();
  });
});
