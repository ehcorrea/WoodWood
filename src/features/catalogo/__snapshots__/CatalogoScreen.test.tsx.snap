// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CatalogoScreen/> quando renderizado e em loading 1`] = `
<ActivityIndicator
  size="large"
  testID="loading"
/>
`;
